src/Workspaces/MSBuildTest/OpenProjectsTests.cs

```diff
@@ -0,0 +1,225 @@
+            {
+                "blazor" or "blazorwasm" or "blazorwasm-empty" =>
+                    [
+                        "CS0246", // The type or namespace name {'csharp_blazor_project'|'App'} could not be found (are you missing a using directive or an assembly reference?)
+                    ],
+                _ => Array.Empty<string>(),
+            };
+
+            await AssertTemplateProjectLoadsCleanlyAsync(templateName, LanguageNames.CSharp, ignoredDiagnostics);
+        }
+
+        [ConditionalTheory(typeof(DotNetSdkMSBuildInstalled))]
+        [MemberData(nameof(GetVisualBasicProjectTemplateNames), DisableDiscoveryEnumeration = false)]
+        public async Task ValidateVisualBasicTemplateProjects(string templateName)
+        {
+            await AssertTemplateProjectLoadsCleanlyAsync(templateName, LanguageNames.VisualBasic);
+        }
+
+        public static TheoryData<string> GetCSharpProjectTemplateNames()
+            => GetProjectTemplateNames("c#");
+
+        public static TheoryData<string> GetVisualBasicProjectTemplateNames()
+            => GetProjectTemplateNames("vb");
+
+        public static TheoryData<string> GetProjectTemplateNames(string language)
+        {
+            // The expected output from the list command is as follows.
+
+            // These templates matched your input: --language='vb', --type='project'
+            //
+            // Template Name                  Short Name           Language  Tags
+            // -----------------------------  -------------------  --------  ---------------
+            // Class Library                  classlib             C#,F#,VB  Common/Library
+            // Console App                    console              C#,F#,VB  Common/Console
+            // ...
+
+            var result = RunDotNet($"new list --type project --language {language}");
+            Assert.Equal(0, result.ExitCode);
+
+            var lines = result.Output
+                .Replace("\r\n", "\n")
+                .Split(new[] { "\n" }, StringSplitOptions.RemoveEmptyEntries);
+
+            TheoryData<string> templateNames = [];
+            var foundDivider = false;
+
+            foreach (var line in lines)
+            {
+                if (!foundDivider)
+                {
+                    if (line.StartsWith("----"))
+                    {
+                        foundDivider = true;
+                    }
+                    continue;
+                }
+
+                var columns = line.Split(new[] { "  " }, StringSplitOptions.RemoveEmptyEntries)
+                    .Select(c => c.Trim())
+                    .ToArray();
+                var templateShortName = columns[1].Split(',').First();
+
+                if (ExcludeMauiTemplates && templateShortName.StartsWith("maui"))
+                    continue;
+
+                templateNames.Add(templateShortName);
+            }
+
+            return templateNames;
```

Comment:
Add an assert here that we did find the divider, lest we end up running no tests at all. Or just generally add an assert that the template names isn't empty.